tiny_gltf.h

```diff
@@ -3279,23 +3347,26 @@ static std::string MimeToExt(const std::string &mimeType) {
   return "";
 }
 
-static bool UpdateImageObject(const Image &image, std::string &baseDir,
+static bool UpdateImageObject(const Image &image,
+                              const std::string &sourceBaseDir, const std::string &targetBaseDir,
                               int index, bool embedImages,
-                              const FsCallbacks *fs_cb,
+                              size_t maxFileSize, const FsCallbacks *fs_cb,
                               const URICallbacks *uri_cb,
                               const WriteImageDataFunction& WriteImageData,
                               void *user_data, std::string *out_uri) {
   std::string filename;
   std::string ext;
-  // If image has uri, use it as a filename
-  if (image.uri.size()) {
-    std::string decoded_uri;
-    if (!uri_cb->decode(image.uri, &decoded_uri, uri_cb->user_data)) {
-      // A decode failure results in a failure to write the gltf.
-      return false;
+  std::string decoded_uri;
+  if (!image.uri.empty()) {
+    // If image has uri, use it as a filename if it is not a data uri
+    if (!IsDataURI(image.uri)) {
```

Comment:
We need to handle data URI here, as we leave images with data URIs as is, instead of loading the data and clearing their URI.